servicetalk-client-api/src/main/java/io/servicetalk/client/api/NoAvailableHostException.java

```diff
@@ -17,11 +17,13 @@
 
 import io.servicetalk.transport.api.RetryableException;
 
+import java.net.SocketException;
+
 /**
  * Thrown when no host is available but at least one is required.
  */
-public class NoAvailableHostException extends RuntimeException implements RetryableException {
-    private static final long serialVersionUID = 5340791072245425967L;
+public class NoAvailableHostException extends SocketException implements RetryableException {
```

Comment:
Is it a "SocketException"? I would rather keep it a runtime one or associate with a LB/SD parent type.

Reply:
The intend was to make it `IOException` because it prevents a request execution due to unavailability to execute an IO operation. During offline discussion, @bondolo recommended a more specific subtype - `SocketException`. I'm agreed with that proposal because it notifies that there are no sockets available at this point of time.

Reply:
Don't see a reason to block the change but I don't agree that this is a good correlation. I agree that `IOException` is a valid super type, but `SocketException` is clearly doc'ed to do with failure to access or create a socket. In our case this is not even related to a connection but rather to host-finding.

Reply:
Ok, I was reading "failure to access a socket" as "no hosts -> no access" 😄 
Changing to `IOException`: https://github.com/apple/servicetalk/pull/1924/commits/c2b41cf0f43e5aaf3ce8e7e4e4cbd8c796ef530f